calyx-frontend/src/parser.rs

```diff
@@ -871,16 +932,32 @@ impl CalyxParser {
         ))
     }
 
+    fn repeat_stmt(input: Node) -> ParseResult<ast::Control> {
+        let span = Self::get_span(&input);
+        Ok(match_nodes!(
+            input.into_children();
+            [at_attributes(attrs), static_word(_), bitwidth(num_repeats) , block(stmt)] => ast::Control::StaticRepeat {
+                num_repeats,
+                body: Box::new(stmt),
+                attributes: attrs.add_span(span),
+            }
+        ))
+    }
+
     fn stmt(input: Node) -> ParseResult<ast::Control> {
         Ok(match_nodes!(
             input.into_children();
             [enable(data)] => data,
             [empty(data)] => data,
             [invoke(data)] => data,
             [seq(data)] => data,
```

Comment:
The benefit of the approach for the `enum` would be that you don't need to have two different parsing variants for each one of the control operators and can maybe parse them in one function